addOns/pscanrulesAlpha/src/test/java/org/zaproxy/zap/extension/pscanrulesAlpha/SubResourceIntegrityAttributeScannerTest.java

```diff
@@ -0,0 +1,157 @@
+                                + "</head><body></body></html>");
+
+        // When
+        rule.scanHttpResponseReceive(msg, -1, createSource(msg));
+
+        // Then
+        assertThat(alertsRaised.get(0).getPluginId(), equalTo(rule.getPluginId()));
+    }
+
+    @Test
+    public void shouldIndicateElementWithoutIntegrityAttribute()
+            throws HttpMalformedHeaderException {
+        // Given
+        HttpMessage msg =
+                buildMessage(
+                        "<html><head>"
+                                + "<script src=\"https://some.cdn.com/v1.0/include.js\"></script>"
+                                + "</head><body></body></html>");
+
+        // When
+        rule.scanHttpResponseReceive(msg, -1, createSource(msg));
+
+        // Then
+        assertThat(
+                alertsRaised.get(0).getEvidence(),
+                equalTo("<script src=\"https://some.cdn.com/v1.0/include.js\"></script>"));
+    }
+
+    @Test
+    public void shouldNotRaiseAlertGivenElementIsNotServedByCDN()
+            throws HttpMalformedHeaderException {
+        // Given
+        HttpMessage msg =
+                buildMessage(
+                        "<html><head>"
+                                + "<script src=\"https://static.example.com/v1.0/include.js\"></script>"
+                                + "<link href=\"http://static.example.com/v1.0/style.css\"></script>"
+                                + "</head><body></body></html>");
+
+        // When
+        rule.scanHttpResponseReceive(msg, -1, createSource(msg));
+
+        // Then
+        assertThat(alertsRaised.size(), equalTo(0));
+    }
+
+    @Test
+    public void shouldNotRaiseAlertGivenElementIsServedByTrustedDomain()
+            throws HttpMalformedHeaderException {
+        // Given
+        HttpMessage msg =
+                buildMessage(
+                        "<html><head>"
+                                + "<script src=\"https://trusted.domain.com/v1.0/include.js\"></script>"
+                                + "<link href=\"https://another.trusted.com/v1.0/include.js\"></link>"
+                                + "</head><body></body></html>");
+        rule.getConfig()
+                .setProperty(
+                        SubResourceIntegrityAttributeScanner.TRUSTED_DOMAINS_PROPERTY,
+                        "https://trusted.domain.com/.*,https://.*trusted.com/.*");
+
+        // When
+        rule.scanHttpResponseReceive(msg, -1, createSource(msg));
+
+        // Then
+        assertThat(alertsRaised.size(), equalTo(0));
+    }
+
+    public HttpMessage buildMessage(String body) throws HttpMalformedHeaderException {
+        HttpMessage msg = new HttpMessage();
+        msg.setRequestHeader("GET http://example.com/ HTTP/1.1");
+        msg.setResponseBody(body);
+        return msg;
+    }
+
+    @Override
+    protected SubResourceIntegrityAttributeScanner createScanner() {
```

Comment:
Could be kept at the beginning for consistency, also better to know how the scanner is being setup for the tests.